### PR TITLE
gh-115119: Switch Windows build to mpdecimal external

### PR DESCRIPTION
**Pull request by zware**
* Issue: gh-115119

### CODE REVIEW

**Comment by zware** [comment]
I missed that `mpdecimal.h` does not actually exist in the source distribution, but is created (or copied) by the building of the library.  This will need to be handled somehow, either by adding a Windows-specific shim that picks the right `mpdecimal{32,64}vc.h` to include somewhere, or by building the library separately using its own build scripts.

**Comment by zware** [comment]
Made it back around to this and fixed the build issue.  I don't like the added `Modules/_decimal/windows/mpdecimal.h` as a permanent solution, but I think it's a decent stopgap to avoid changing the sources somehow.  Ideally we'll eventually switch to using `mpdecimal`'s own PGO build script to produce binaries, but that will be a bigger change that we can make happen later.  It will also require some attention on ARM, unless 4.0.0 already has handling for that.

**Comment by zware** [comment]
!buildbot Windows

File: PCbuild/_decimal.vcxproj [hunk 1]
```diff
@@ -93,51 +93,55 @@
...
     </ClCompile>
   </ItemDefinitionGroup>
   <ItemGroup>
-    <ClInclude Include="..\Modules\_decimal\libmpdec\basearith.h" />
-    <ClInclude Include="..\Modules\_decimal\libmpdec\bits.h" />
-    <ClInclude Include="..\Modules\_decimal\libmpdec\constants.h" />
-    <ClInclude Include="..\Modules\_decimal\libmpdec\convolute.h" />
-    <ClInclude Include="..\Modules\_decimal\libmpdec\crt.h" />
-    <ClInclude Include="..\Modules\_decimal\libmpdec\difradix2.h" />
+    <ClInclude Include="$(mpdecimalDir)\libmpdec\basearith.h" />
```
Comment (zooba) [inline PCbuild/_decimal.vcxproj:105]: I don't think this is going to play well with Visual Studio (similarly for the filters file). Would be worth opening it up and seeing how the files are grouped to see if it's usable.

Might be best off making these ones regular filenames and then using the IDE to arrange them, then revert its changes to this file (if any) and keep the updated filters file.

Comment (zooba) [reply PCbuild/_decimal.vcxproj:105]: Oh except these are being pulled from Externals now, so those aren't going to be stable names.

I _think_ there's a metadata option (sub element) to make these be treated as links, which might solve things enough. Easiest way to find that will be to add the file as a link in VS, save it, and then copy the extra XML into each item. I think it's a `<Link>fake path inside project</Link>` element.

Comment (zware) [reply PCbuild/_decimal.vcxproj:105]: As far as I can tell, VS is happy with this as is; at least I'm not seeing anything screaming about it, anyway :).  I don't know if I'm looking in the right place, though.

There's definitely room to clean this up (ideally building `libmpdec` separately using its own facilities and just linking it in), but I think I'd like to aim small for this first step.

**Comment by zooba** [comment]
@zware This looks like it's good enough shape - any reason to not merge yet?

**Comment by zware** [comment]
The limitation of 24h/day, mostly :)

**Pull request merged**